hpat/tests/tests_perf/test_perf_series_str.py

```diff
@@ -0,0 +1,383 @@
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import itertools
+import os
+import time
+import unittest
+from contextlib import contextmanager
+
+import pandas as pd
+
+import hpat
+from hpat.tests.tests_perf.test_perf_utils import *
+
+
+STRIP_CASES = [
+    'ascii',
+    'tú quiénc te crees?',
+    '大处 着眼，c小处着手c。大大c大处'
+]
+
+
+def usecase_series_len(input_data):
+    start_time = time.time()
+
+    input_data.str.len()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_capitalize(input_data):
+    start_time = time.time()
+
+    input_data.str.capitalize()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lower(input_data):
+    start_time = time.time()
+
+    input_data.str.lower()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_swapcase(input_data):
+    start_time = time.time()
+
+    input_data.str.swapcase()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_title(input_data):
+    start_time = time.time()
+
+    input_data.str.title()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_upper(input_data):
+    start_time = time.time()
+
+    input_data.str.upper()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.lstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_rstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.rstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_strip(input_data):
+    start_time = time.time()
+
+    input_data.str.strip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+@contextmanager
+def do_jit(f):
+    """Context manager to jit function"""
+    cfunc = hpat.jit(f)
+    try:
+        yield cfunc
+    finally:
+        del cfunc
+
+
+def calc_time(f, data):
+    """Calculate execution time of specified function."""
+    start_time = time.time()
+    f(data)
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def calc_compile_time(f, data):
+    """Calculate compile time as difference between first 2 runs."""
+    return calc_time(f, data) - calc_time(f, data)
+
+
+def calc_compilation(pyfunc, data):
+    """Calculate compile time several times."""
+    compile_iteration_number = 5
+
+    compile_times = []
+    for _ in range(compile_iteration_number):
+        with do_jit(pyfunc) as cfunc:
+            compile_times.append(calc_compile_time(cfunc, data))
```

Comment:
Could you please avoid writing diffrent sentence in one line? Please, don't borrow "bad thing" from python.